### PR TITLE
Haxe LSP Lite-XL plugin

### PR DESCRIPTION
**Pull request by SnakE-2024**
Second part of Haxe LSP support

### CODE REVIEW

**Review by Guldoman — COMMENTED** [review]
What do you think about also bundling haxe itself? It seems like the lsp server allows specifying a custom executable, and the haxe website has some downloads we should be able to use.

**Comment by SnakE-2024** [comment]
I thought about it. Let me test it. And if it will be Ok, I'll create yet another PR

**Comment by Guldoman** [comment]
Feel free to do it in this one. It doesn't even have to be a separate "plugin" as at the moment nothing else would use it.

File: manifest.json
Comment (Guldoman) [inline manifest.json:1]: Could you sort the new entries alphabetically in the file?

**Review by Guldoman — COMMENTED** [review]
We're nearly there, thank you for your patience.

File: plugins/lsp_haxe.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
+-- mod-version:3
+
+local lsp = require("plugins.lsp")
+local lspconfig = require("plugins.lsp.config")
+local common = require("core.common")
+local config = require("core.config")
+local core = require("core")
+
+local node = require("libraries.nodejs")
+local haxe = require("libraries.haxe")
```
[the remark this answers is not in the document]
Comment (SnakE-2024) [reply plugins/lsp_haxe.lua:10]: Sure

File: README.md
[the remark this answers is not in the document]
Comment (SnakE-2024) [reply README.md:1]: What section do you mean?

**Comment by SnakE-2024** [comment]
All requested changes have been made

**Review by Guldoman — COMMENTED** [review]
And obviously we encounter a bug with the server...
It looks like we still need
```lua
  settings = {
    haxe = {
    },
  },
```
even if empty, or it just won't work.

Btw you can test this PR with `lpm run --ephemeral https://github.com/SnakE-2024/lite-xl-lsp-servers:lsp_haxe_plugin lsp_haxe`.

**Comment by Guldoman** [comment]
Without it, I'm getting a "Method not found" error as reply to our "textDocument/didOpen" for some reason...

**Comment by Guldoman** [comment]
Thank you!

**Comment by Guldoman** [comment]
Next step is to add the stub to both `lsp_haxe` and `haxe` in our plugins repo.

**Comment by SnakE-2024** [comment]
Thank you!
So, what are my next actions?

**Comment by Guldoman** [comment]
Have a look at https://github.com/lite-xl/lite-xl-plugins/pull/376 for an example.

**Pull request merged**